ddcommon-ffi/src/error.rs

```diff
@@ -111,3 +111,15 @@ pub type MaybeError = crate::Option<Error>;
 
 #[no_mangle]
 pub extern "C" fn ddog_MaybeError_drop(_: MaybeError) {}
+
+#[macro_export]
+macro_rules! try_c {
```

Comment:
Does this have to be a macro? Would something like this work instead?
```rust
pub fn try_c<T, E: std::fmt::Debug>(failable: Result<T, E>) -> Result<T, MaybeError> {
    match failable {
        Ok(o) => Ok(o),
        Err(e) => Err(MaybeError::Some(Error::from(format!("{:?}", e)))),
    }
}
```

Reply:
The point of try_c() is to return within a function returning MaybeError. The conversion itself, to MaybeError::Some isn't the important part here.

Reply:
@bwoebi I'm not sure I understand. My comment is asking if we _need_ to use a macro here versus a generic function. Personally, I'm not a fan of macros unless they are truly necessary / really the best option.